app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php

```diff
@@ -28,35 +30,62 @@ class GetStockItemData implements GetStockItemDataInterface
      */
     private $stockIndexTableNameResolver;
 
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * @var GetProductIdsBySkusInterface
+     */
+    private $getProductIdsBySkus;
+
     /**
      * @param ResourceConnection $resource
      * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     * @param GetProductIdsBySkusInterface $getProductIdsBySkus
      */
     public function __construct(
         ResourceConnection $resource,
-        StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider,
+        GetProductIdsBySkusInterface $getProductIdsBySkus
     ) {
         $this->resource = $resource;
         $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->defaultStockProvider = $defaultStockProvider;
+        $this->getProductIdsBySkus = $getProductIdsBySkus;
     }
 
     /**
      * @inheritdoc
      */
     public function execute(string $sku, int $stockId): ?array
     {
-        $stockItemTableName = $this->stockIndexTableNameResolver->execute($stockId);
-
         $connection = $this->resource->getConnection();
-        $select = $connection->select()
-            ->from(
+        $select = $connection->select();
+
+        if ($this->defaultStockProvider->getId() === $stockId) {
+            $productId = current($this->getProductIdsBySkus->execute([$sku]));
+            $stockItemTableName = $this->resource->getTableName('cataloginventory_stock_status');
+            $select->from(
+                $stockItemTableName,
+                [
+                    GetStockItemDataInterface::QUANTITY => 'qty',
```

Comment:
Constants may be used 
\Magento\CatalogInventory\Api\Data\StockStatusInterface::QTY
\Magento\CatalogInventory\Api\Data\StockStatusInterface::STOCK_STATUS

Reply:
Solution requires adding a dependency on the `CatalogInventory` this should be done in a centralized way, so filed a separate ticket for it.
https://github.com/magento/inventory/issues/2387